File: Problems/Week_2/3.29_큐(10845)/서봉성/10845.cpp
#include<iostream>
#include<string>

using namespace std;

int main(void) {
	int n1, n2, n3 = 1, top = -1, bottom = 0;
	string str;
	int queue[10000];
	cin >> n1;
	while (n3++ <= n1) {
		cin >> str;
		if (str == "push") {
			cin >> n2;
			queue[++top] = n2;
		}
		else if (str == "pop") {
			if (top - bottom == -1)
				cout << -1 << endl;
			else
				cout << queue[bottom++] << endl;
		}
		else if (str == "size") {
			cout << top - bottom + 1 << endl;
		}
		else if (str == "empty") {
			if (top - bottom == -1)
				cout << 1 << endl;
			else
				cout << 0 << endl;
		}
		else if (str == "front") {
			if (top - bottom == -1)
				cout << -1 << endl;
			else
				cout << queue[bottom] << endl;
		}
		else if (str == "back") {
			if (top - bottom == -1)
				cout << -1 << endl;
			else
				cout << queue[top] << endl;
		}
	}

	return 0;

}

File: Problems/Week_2/3.29_큐(10845)/이승민/10845.cs
using System;
using System.Text;

namespace 백준1
{
    class mainprogram
    {
        static void Main(string[] args)
        {
            int n = Int32.Parse(Console.ReadLine());
            int[] queue = new int[n];
            short queuepointer = 0; //스택은 데이터의 출입이지만 여기선 들어갈 자리만 해당함
            StringBuilder sb = new StringBuilder();

            while (n-- != 0)
            {
                string[] command = Console.ReadLine().Split(' ');

                switch (command[0])
                {
                    case "push":
                        queue[queuepointer++] = Int32.Parse(command[1]);
                        break;
                    case "pop":
                        if (queuepointer == 0)
                        {
                            sb.Append(-1 + "\n");
                        }
                        else
                        {
                            sb.Append(queue[0] + "\n");
                            for (int i = 0; i < queuepointer; i++)
                            {
                                queue[i] = queue[i + 1];
                            }
                            --queuepointer;

                        }
                        break;
                    case "size":
                        sb.Append(queuepointer + "\n");
                        break;
                    case "empty":
                        if (queuepointer == 0)
                        {
                            sb.Append(1 + "\n");
                        }
                        else
                        {
                            sb.Append(0 + "\n");
                        }
                        break;
                    case "front":
                        if (queuepointer == 0)
                        {
                            sb.Append(-1 + "\n");
                        }
                        else
                        {
                            sb.Append(queue[0] + "\n");
                        }
                        break;
                    case "back":
                        if (queuepointer == 0)
                        {
                            sb.Append(-1 + "\n");
                        }
                        else
                        {
                            sb.Append(queue[queuepointer-1] + "\n");
                        }
                        break;
                }


            }

            Console.WriteLine(sb);

        }
    }
}
